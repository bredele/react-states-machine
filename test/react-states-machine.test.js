/**
 * Dependencie(s)
 */

import React from 'react'
import machine from '../lib'
import renderer from 'react-test-renderer'


test('should render first state', () => {
  const flow = renderer.create(machine({
    'init': [
      props => <button>hello world</button>
    ]
  }))
  let tree = flow.toJSON()
  expect(tree).toMatchSnapshot()
})


test('should render given state', () => {
  const flow = renderer.create(machine({
    'init': [
      props => <button>hello world</button>
    ],
    'other': [
      props => <button>other</button>
    ]
  }, {}, 'other'))
  let tree = flow.toJSON()
  expect(tree).toMatchSnapshot()
})


test('should go to given state', () => {
  const flow = renderer.create(machine({
    'init': [
      props => <button onClick={() => props.goto('other')}>hello world</button>
    ],
    'other': [
      props => <button>other</button>
    ]
  }))

  let tree = flow.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()


  tree = flow.toJSON()
  expect(tree).toMatchSnapshot()
})

test('should go to given state and pass props', () => {
  const flow = renderer.create(machine({
    'init': [
      props => <button onClick={() => props.goto('other', {
        label: 'what'
      })}>hello world</button>
    ],
    'other': [
      props => <button>props.label</button>
    ]
  }))

  let tree = flow.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()


  tree = flow.toJSON()
  expect(tree).toMatchSnapshot()
})


test('should transition current state and update props', () => {
  const flow = renderer.create(machine({
    'init': [
      props => <button onClick={() => props.transition('john')}>hello {props.message || 'world'}</button>,
      {
        'john': [() => {
          return {
            message: 'john'
          }
        }]
      }
    ]
  }))

  let tree = flow.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()

  tree = flow.toJSON()
  expect(tree).toMatchSnapshot()
})


test('should transition to other state and pass props', () => {
  const flow = renderer.create(machine({
    'init': [
      props => <button onClick={() => props.transition('next')}>hello world</button>,
      {
        'next': [() => ({message: 'other'}), 'other']
      }
    ],
    'other': [
      props => <span>{props.message}</span>
    ]
  }))

  let tree = flow.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()


  tree = flow.toJSON()
  expect(tree).toMatchSnapshot()
})
