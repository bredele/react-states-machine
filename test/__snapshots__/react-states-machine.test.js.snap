// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should go to given state 1`] = `
<button
  onClick={[Function]}
>
  hello world
</button>
`;

exports[`should go to given state 2`] = `
<button>
  other
</button>
`;

exports[`should go to given state and pass props 1`] = `
<button
  onClick={[Function]}
>
  hello world
</button>
`;

exports[`should go to given state and pass props 2`] = `
<button>
  props.label
</button>
`;

exports[`should render first state 1`] = `
<button>
  hello world
</button>
`;

exports[`should render given state 1`] = `
<button>
  other
</button>
`;

exports[`should transition current state 1`] = `
<button
  onClick={[Function]}
>
  hello
  world
</button>
`;

exports[`should transition current state 2`] = `
<button
  onClick={[Function]}
>
  hello
  john
</button>
`;

exports[`should transition current state and update props 1`] = `
<button
  onClick={[Function]}
>
  hello 
  world
</button>
`;

exports[`should transition current state and update props 2`] = `
<button
  onClick={[Function]}
>
  hello 
  world
</button>
`;

exports[`should transition to other state 1`] = `
<button
  onClick={[Function]}
>
  hello world
</button>
`;

exports[`should transition to other state 2`] = `
<span>
  other
</span>
`;

exports[`should transition to other state and pass props 1`] = `
<button
  onClick={[Function]}
>
  hello world
</button>
`;

exports[`should transition to other state and pass props 2`] = `
<button
  onClick={[Function]}
>
  hello world
</button>
`;
